cmds/find/find.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Find finds files. It is similar to the Unix command. It uses REs, not globs,
+// for matching.
+//
+// OPTIONS:
+//     -d: enable debugging in the find package
+//     -mode integer-arg: match against mode, e.g. -mode 0755
+//     -type: match against a file type, e.g. -type f will match files
+//     -name: name is an RE, e.g. -name '.*[a-z][0-9]$' will match files ending in a letter and number.
+//            This is very useful for /dev
+//     -l: long listing. It's not very good, yet, but it's useful enough.
+//     -root: where to start the search, default /tmp
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/find"
+)
+
+var (
+	perm      = flag.Int("mode", -1, "Permissions")
+	fileType  = flag.String("type", "", "File type")
+	name      = flag.String("name", ".*", "re for name")
```

Comment:
I suggest have the default name as an empty string. Then conditionally skip doing the regex match when name is an empty string in func (f *Finder) Find(), i.e. push all results into the Name chan. This skips doing extra work of checking a regex that always matches for the default use case.

Reply:
I would also spell out "regular expression" instead of "re" in the help string for name, so it's clear we're doing something different than coreutils find.

Reply:
OK, I left the door open in the package for using things other than glob, but the command itself uses a glob.